### ISSUE TITLE
Lfd

### ISSUE THREAD

**Issue by swift1988** [OSVAI/Af-DCD#1]
When Lfd is removed, the mIoU is very low.

Or can you upload the models distilled without Lfd?

**Comment by jwfanDL** [comment]
Thank you for your interests on our paper.

You can refer our statement in Section 3.2 and results in Table 3(a) and Table 3(b) to address your concerns.

We answer two key questions in  Section 3.2 and draw a conclusion that FD and Af-DCD  are inextricably bound up with each other.  If you directly remove FD, the gain of the method may be quite low in Cityscapes (shown in Table 3(a)). 

However, we observe very different results on ADE20K (in Table 3(b)). This may result from different distribution of different datasets or resolution.

If you want to observe and study the gain from Af-DCD loss, we recommend you to deploy on ADE20K dataset.

**Comment by swift1988** [comment]
Considering the Af-dcd executed on the masked feature, if I understand correctly, the Af-dcd will introduce too much noisy information into the training without FD. Distilling the student with FD can alleviate this problem.

Regarding comparison:  The FD is the same as MGD, which already shows competitive results on distilling segmenters.
Adding FD/MGD to other KD methods (e.g., MasKD, CWD) is a more fair comparison.

I noticed that the Af-dcd works differently in different datasets. One possible reason, I think, is that KD is quite sensitive to teacher-student pairs, datasets, hyper-parameters, etc. 

Anyway, great work, and thanks for your reply.

**Comment by jwfanDL** [comment]
Thank you for your comments. According to your comments, we'd like to clarify your two questions.

**(1) Whether Af-DCD introduce too much "noisy information"?** 
(a) Maybe not.  So-called "noisy information" indicates that these extra information does harm to KD process. (b) However, our Af-DCD is a contrastive loss defined based on teacher-student feature pairs. In terms of mathematical definition, our designed contrastive term is a relaxation form of self-similarity mimicking.  (c) Therefore, the reason why Af-DCD shows low gain on Cityscapes is that this term is not strong enough, but not introducing extra noisy information. 

**(2) Fair comparison to MasKD, CWD and etc.**
(a) Your question may be resulted from misunderstanding of our line of research. (b) No matter MasKD and CWD, their designs are to improve FD. For instance, MasKD leverages instance-level masks to enhance the information related to foreground. (b) Comparing to these methods, our FD is much simpler.  Thus, adding FD to MasKD or other methods seems not reasonable. (c) On the other hand, our method can be implemented on MasKD. These kinds of experiments can further show the generalization of our method.  We may consider implement in the future.

**Comment by swift1988** [comment]
Thanks for your reply and clarifications. I would like to discuss that:

(1) "noisy information"
(a)   The "noisy information" does not necessarily make the loss term harm the student. For example, distilling the detectors with full feature mimicking can improve (although slightly) the student. However, the massive background regions are still 'noisy" when distilling detectors.
(b)   According to your code, the mask ratio is 0.75. We may test if there is ''noisy information" by changing the value of the mask ratio. A guess: decreasing the mask ratio will facilitate the Af-dcd.

(2)   Fair comparison
(a) By 'fair' here, I mean the FD is the same as MGD, which is a strong baseline. In this case, FD+CWD v.s. FD+MasKD v.s. FD+Af-dcd may be a reasonable comparison. Besides, it seems that you change the original generator in MGD with another "EnhanceProjector".
(b) Besides, As stated in Sec. 3.3, the Af-dcd and FD are inextricably bound up with each other. Particularly, you mentioned that "… Af-dcd offers FD essential constraints on …". So, one may wonder if other relation-based knowledge, like the affinity map in SKD and the intra-class variation map in IFVD, would also offer the desired effect. After all, the complementary of feature-based and relation-based knowledge has been verified.

**Comment by YaoAnbang** [comment]
Really insightful thought! You may run some experiments with our code to validate your guess.
